tests/hlsl/register-syntax.slang

```diff
@@ -0,0 +1,48 @@
+//TEST:SIMPLE(filecheck=CHECK_GLSL): -target glsl -stage compute -entry computeMain
+//TEST:SIMPLE(filecheck=CHECK_HLSL): -target hlsl -stage compute -entry computeMain
+//CHECK_HLSL: u4, space2
+//CHECK_HLSL: u5, space1
+//CHECK_HLSL: u6
+//CHECK_HLSL: u7
+//CHECK_HLSL: u9
+//CHECK_HLSL: u12
+
+//CHECK_GLSL: binding = 4, set = 2
+//CHECK_GLSL: binding = 5, set = 1
+//CHECK_GLSL: binding = 6
+//CHECK_GLSL: binding = 7
+//CHECK_GLSL: binding = 10
+//CHECK_GLSL: binding = 12
```

Comment:
The order of the variables may change.
You should include the variable name in the search pattern like,
```
//CHECK_HLSL: b0{{.*}}u4, space2
//CHECK_HLSL: b1{{.*}}u5, space1

Reply:
very good point about the order, I think I need to use `CHECK-DAG` though since `CHECK` enforces ordering of statements. This is an issue since if `b0` and `b1` are swapped, `CHECK` would not find the statements.